GPU/Software/Rasterizer.cpp

```diff
@@ -114,11 +114,11 @@ static inline u8 ClampFogDepth(float fogdepth) {
 
 static inline void GetTexelCoordinates(int level, float s, float t, int& out_u, int& out_v)
 {
-	int width = 1 << (gstate.texsize[level] & 0xf);
-	int height = 1 << ((gstate.texsize[level]>>8) & 0xf);
+	int width = gstate.getTextureWidth(level);
+	int height = gstate.getTextureHeight(level);
 
-	int u = (int)(s * width);
-	int v = (int)(t * height);
+	int u = (int)(s * width + 0.375f);
+	int v = (int)(t * height + 0.375f);
```

Comment:
Well, AFAICT it's just 0 for nearest (seems to round right at 0.5) and -0.5 for linear (to sample at the centers.)  Need to check some throughmode handling still when I have time (we currently truncate float uvs to integer in SW and I'm not sure that's right.)

I did try applying a `0.375f / 256.0f` offset to correct for rounding issues, which seemed to largely help most everywhere (including Crisis Core.)  And with that the Hexyz issue doesn't happen either (probably the same rounding issue.)

I think the biggest remaining problem (not sure if introduced recently) is triangles not "lining up" and either overlapping or having 1px between (like the Crisis Core pause screen.)  Probably a top left bias problem, but not sure.

-[Unknown]